docs/orchestrating-elastic-stack-applications/elasticsearch/jvm-heap-dumps.asciidoc

```diff
@@ -20,15 +20,15 @@ To take a heap dump before the JVM process runs out of memory you can execute th
 [source,sh,subs="attributes,+macros"]
 ----
 kubectl exec $POD_NAME -- su elasticsearch -g root -c \
-  '/usr/share/elasticsearch/jdk/bin/jmap -dump:format=b,file=data/heap.hprof $(pgrep java)'
+  '/usr/share/elasticsearch/jdk/bin/jmap -dump:format=b,file=data/heap.hprof $(pgrep -n java)'
```

Comment:
After some digging, using jps is probably the most reliable to get the PID:
```sh
# elasticsearch@quickstart-es-default-0
> /usr/share/elasticsearch/jdk/bin/jps -l | grep org.elasticsearch.bootstrap.Elasticsearch | cut -f1 -d" "
68
```

I noticed other issues: `su elasticsearch` and `-g root` are no longer to be used.

```
> kubectl exec $POD_NAME -- su elasticsearch -g root -c \
  '/usr/share/elasticsearch/jdk/bin/jmap -dump:format=b,file=data/heap.hprof \
  $(/usr/share/elasticsearch/jdk/bin/jps -l | grep org.elasticsearch.bootstrap.Elasticsearch | cut -f1 -d" ")'

su: only root can specify alternative groups

/u/k/d/s/g/e/elasticsearch-controller updating-command-to-extract-es-pid * krkr@fun
> kubectl exec $POD_NAME -- su elasticsearch  -c \
  '/usr/share/elasticsearch/jdk/bin/jmap -dump:format=b,file=data/heap.hprof \
  $(/usr/share/elasticsearch/jdk/bin/jps -l | grep org.elasticsearch.bootstrap.Elasticsearch | cut -f1 -d" ")'

Password: su: Authentication failure
```

The final command that works is:
```
> kubectl exec $POD_NAME -- bash -c \
  '/usr/share/elasticsearch/jdk/bin/jmap -dump:format=b,file=data/heap.hprof \
  $(/usr/share/elasticsearch/jdk/bin/jps -l | grep org.elasticsearch.bootstrap.Elasticsearch | cut -f1 -d" ")'
Defaulted container "elasticsearch" out of: elasticsearch, elastic-internal-init-filesystem (init), elastic-internal-suspend (init)
Dumping heap to /usr/share/elasticsearch/data/heap.hprof ...
Heap dump file created [423089695 bytes in 6.035 secs]
```
But we need to double check it is okay for older ES versions.

Reply:
In versions 7 and 8 it's ok but in version 6 the jdk is in `/opt/jdk-15.0.1+9/bin/`. We were already documenting the use of `/usr/share/elasticsearch/jdk/bin/jmap` so it was already broken for version 6.

Either we go with the "solid" (but verbose) `/usr/share/elasticsearch/jdk/bin/jps -l | grep org.elasticsearch.bootstrap.Elasticsearch | cut -f1 -d" "`, that does not work in 6 ; or we go with `pgrep -n java` which should be ok in 99.9?% of cases.